WordPress/src/main/java/org/wordpress/android/ui/reader/viewmodels/ReaderPostDetailViewModel.kt

```diff
@@ -195,6 +209,24 @@ class ReaderPostDetailViewModel @Inject constructor(
         }
     }
 
+    private fun onRelatedPostItemClicked(postId: Long, blogId: Long, isGlobal: Boolean) {
+        trackRelatedPostClickAction(postId, blogId, isGlobal)
+        _navigationEvents.value = if (isRelatedPost) {
+            Event(ReplaceRelatedPostDetailsWithHistory(postId = postId, blogId = blogId, isGlobal = isGlobal))
+        } else {
+            Event(ShowRelatedPostDetails(postId = postId, blogId = blogId))
+        }
+    }
+
+    private fun onRelatedPostFollowClicked(relatedPost: ReaderSimplePost) {
```

Comment:
Suggestion (💡): Since you only need the `sideId` and `siteName`, maybe passing only those instead of the whole `post` object will be better? Consider the fact that you already did the same for the `onRelatedPostItemClicked(...)` function, instead of passing the whole `state` object.

Reply:
Makes sense 👍. Even though we'll remove follow button as per updated mockup and will not require this method, for the completeness of this PR, I included this suggestion in 843d321.